kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/ProjectSourceClass.java

```diff
@@ -42,32 +54,43 @@ public ProjectSourceClass withDependencyInjection(String dependencyInjection) {
     }
 
     public String generate() {
-        StringBuilder sb = new StringBuilder();
-        sb.append(
-                "package org.drools.project.model;\n" +
-                        "\n" +
-                        "import " + KieBase.class.getCanonicalName() + ";\n" +
-                        "import " + KieBaseModel.class.getCanonicalName() + ";\n" +
-                        "import " + KieSession.class.getCanonicalName() + ";\n" +
-                        "import " + KieBaseBuilder.class.getCanonicalName() + ";\n" +
-                        "\n" +
-                        dependencyInjection + "\n" +
-                        "public class ProjectRuntime implements " + KieRuntimeBuilder.class.getCanonicalName() + " {\n" +
-                        "\n" +
-                        "    private final ProjectModel model = new ProjectModel();\n" +
-                        "    private final java.util.Map<String, KieBase> kbases = new java.util.HashMap<>();\n" +
-                        "\n" +
-                        "    public static final ProjectRuntime INSTANCE = new ProjectRuntime();\n" +
-                        "\n");
-        sb.append(modelMethod.toGetKieBaseMethods());
-        sb.append("\n");
-        sb.append(modelMethod.toNewKieSessionMethods());
-        sb.append("\n");
-        sb.append(modelMethod.toGetKieBaseForSessionMethod());
-        sb.append("\n");
-        sb.append(modelMethod.toKieSessionConfMethod());
-        sb.append("\n}");
-        return sb.toString();
+        TemplatedGenerator generator = TemplatedGenerator.builder()
+                .withTemplateBasePath(TEMPLATE_RULE_FOLDER)
+                .withPackageName("org.drools.project.model")
+                .withFallbackContext(JavaKogitoBuildContext.CONTEXT_NAME)
+                .build(context, "ProjectRuntime");
+
+        CompilationUnit cu = generator.compilationUnitOrThrow("Could not create CompilationUnit");
+        ClassOrInterfaceDeclaration clazz = cu
+                .findFirst(ClassOrInterfaceDeclaration.class)
+                .orElseThrow(() -> new NoSuchElementException("Compilation unit doesn't contain a class or interface declaration!"));
+
+        if (dependencyInjection != null && dependencyInjection.length() > 0) {
+            clazz.addAnnotation(dependencyInjection);
+        }
```

Comment:
Please remove `withDependencyInjection` method from this class (and corresponding invocation in `IncrementalRuleCodegen`) and replace this with
```suggestion
        if (context.hasDI()) {
            context.getDependencyInjectionAnnotator().withApplicationComponent(clazz);
        }
```